docs/user/troubleshooting: Add generic advice for getting pod details

wking opened the pull request:
Because this is the next step beyond "my cluster didn't install, here's the `--log-level=debug` output", for folks working up bug reports that don't match one of the common failures.

CC @crawford

wking commented on docs/user/troubleshooting.md:
````diff
@@ -89,3 +89,45 @@ This is safe to ignore and merely indicates that the etcd bootstrapping is still
+...
+```
+
+Any nodes are listed as `Pending`, you can investigate with:
````
> pods

Thanks; fixed with ce7e80a -> c38c606.

abhinavdahiya commented:
This is now bleeding into debugging k8s. i don't think this repo should be the source for that. i would say just point to https://kubernetes.io/docs/tasks/debug-application-cluster/ for debugging

abhinavdahiya commented on docs/user/troubleshooting.md:
````diff
@@ -89,3 +89,45 @@ This is safe to ignore and merely indicates that the etcd bootstrapping is still
+Warning  FailedScheduling  1m (x10 over 1m)  default-scheduler  0/1 nodes are available: 1 node(s) had taints that the pod didn't tolerate.
+```
+
+### Check for Unready Nodes
````
the describe on node will give you conditions for the node object that will provide information as to why the node is not ready. it might be network not ready, disk pressure etc.. again i think we should point to the docs on kubelet node conditions.

abhinavdahiya replied:
https://kubernetes.io/docs/concepts/architecture/nodes/#node-status

wking replied:
> the describe on node will give you conditions...

Obsolete now that I've [dropped this section][1].

[1]: https://github.com/openshift/installer/pull/664#issuecomment-438425041

wking commented:
> This is now bleeding into debugging k8s...

Good point.  With c38c606 -> 8ba79d5, I've dropped the node section and added a link to those upstream docs.  I would like to keep the pod section, because I can never remember the new-in-8ba79d5 incantation to get from pods -> image repo/commit.  Thoughts?

abhinavdahiya commented:
> because I can never remember the new-in-8ba79d5 incantation to get from pods -> image repo/commit. Thoughts?

i think we should be using `oc adm release info quay.io/openshift-release-dev/ocp-release:4.0.0-2 --commits` to get information regarding the images based on the release image used.

abhinavdahiya commented:
> Why quay.io?
it was just an example :/

the best way to release image is going to be `oc get clusterversion version`. but i don't think it shows the correct version yet...

wking commented:
> the best way to release image is going to be `oc get clusterversion version`. but i don't think it shows the correct version yet...

openshift/cluster-version-operator#51 moves us in this direction, but do we need to block this PR on that?

abhinavdahiya commented on docs/user/troubleshooting.md:
```diff
@@ -89,3 +89,59 @@ This is safe to ignore and merely indicates that the etcd bootstrapping is still
+
+### Check for Pending or Crashing Pods
+
+This is the generic version of the [*No Worker Nodes Created*](#no-worker-nodes-created) troubleshooting procedure.
```
this is already part of `## Generic Troubleshooting` why explain again?

wking replied:
> this is already part of `## Generic Troubleshooting` why explain again?

I wanted to link back to the *No Worker Nodes Created* section.  Can you float alternative phrasing for that link?

abhinavdahiya commented on docs/user/troubleshooting.md:
````diff
@@ -89,3 +89,59 @@ This is safe to ignore and merely indicates that the etcd bootstrapping is still
+This is the generic version of the [*No Worker Nodes Created*](#no-worker-nodes-created) troubleshooting procedure.
+
+```console
+$ oc --config=${INSTALL_DIR}/auth/kubeconfig get pods --all-namespaces
````
what is this ${INSTALL_DIR} ? can we point to something that setups the kubeconfig

wking replied:
> what is this `${INSTALL_DIR}`?

What you passed to `--dir` when calling `openshift-install create ...`.  That's currently implied by the name, but not documented explicitly.

> can we point to something that setups the kubeconfig

In this PR, I'm just rolling forward with the pattern that [already exists in master][1], so if you'd prefer an explicit introduction to it, it could go in via a parallel PR without conflicting with this one.

[1]: https://github.com/openshift/installer/blame/32a15a5330cb39b8256634ea4d1f98e160e7328f/docs/user/troubleshooting.md#L14

abhinavdahiya commented on docs/user/troubleshooting.md:
````diff
@@ -89,3 +89,59 @@ This is safe to ignore and merely indicates that the etcd bootstrapping is still
+And you can see where that image comes from with:
+
+```console
+$ oc --config=${INSTALL_DIR}/auth/kubeconfig adm release info registry.svc.ci.openshift.org/openshift/origin-release:v4.0-20181113175638 --commits
````
this should not be run with --config as we are not talking to our cluster.

wking replied:
> this should not be run with `--config`...

Fixed with a03bd6b -> a40cdc1.

abhinavdahiya commented:
> > the best way to release image is going to be `oc get clusterversion version`. but i don't think it shows the correct version yet...
> 
> [openshift/cluster-version-operator#51](https://github.com/openshift/cluster-version-operator/pull/51) moves us in this direction, but do we need to block this PR on that?

no, we can continue with what we have right now

abhinavdahiya commented:
i think this is fine for now.

/lgtm
